### PR TITLE
Explanation on steps,source,and destination nodes

### LINKED ISSUE #7504: Explain best practices for sample files
# General issue

## Description:
The migration tool contains a bunch of sample files for the mapping, settings, etc that we can copy to our own module for configuring the migration tool.  The map.xml.dist file, for instance, contains over 2200 lines with hundreds of entries.  Is this just an example or are we supposed to keep all that stuff in there and add our own customizations as needed?  This is not explained at all.  It tells you to use the files included in the module, but not what content within the file should be kept.

https://devdocs.magento.com/guides/v2.3/migration/migration-tool-internal-spec.html

Reference similar closed issue for more context: https://github.com/magento/devdocs/issues/7503

## Possible solutions:
You need to explain better how we're supposed to use the sample files.  Should they be left as is and only added to?  Or should we wipe it all out and just put in the things we want to put in?  I don't even know what half those sample entries are, let alone if they're important or should be ignored.

### PR DESCRIPTION
**Pull request by ajithkumar-maragathavel**
## Purpose of this pull request

This pull request (PR) adds an explanation to the steps, source, and destination nodes.

## Affected DevDocs pages

-  https://devdocs.magento.com/guides/v2.4/migration/migration-tool-internal-spec.html
-  https://devdocs.magento.com/guides/v2.3/migration/migration-tool-internal-spec.html

## Links to Magento source code

- https://github.com/magento/data-migration-tool/blob/2.4-develop/etc/config.xsd

Fixes #7504

### CODE REVIEW

File: src/guides/v2.3/migration/migration-tool-internal-spec.md [hunk 1]
```diff
@@ -165,6 +165,42 @@ Change prefix option in case Magento was installed with prefix in database table
 
 Configuration data is accessible via \Migration\Config class.
 
+### steps available operations {#steps}
```
Comment (BarnyShergold) [inline src/guides/v2.3/migration/migration-tool-internal-spec.md:168]:
```suggestion
### Steps available operations {#steps}
```

File: src/guides/v2.3/migration/migration-tool-internal-spec.md [hunk 2]
```diff
@@ -165,6 +165,42 @@ Change prefix option in case Magento was installed with prefix in database table
...
+
+|Document|Field|
+|--- |--- |
```
Comment (BarnyShergold) [inline src/guides/v2.3/migration/migration-tool-internal-spec.md:171]:
```suggestion
|---|---|
```

File: src/guides/v2.3/migration/migration-tool-internal-spec.md [hunk 3]
```diff
@@ -165,6 +165,42 @@ Change prefix option in case Magento was installed with prefix in database table
...
+|step|It is the second level node inside the Steps node. It provides title to the relavent steps.|
```
Comment (BarnyShergold) [inline src/guides/v2.3/migration/migration-tool-internal-spec.md:172]:
```suggestion
|step|Second level node inside the Steps node. Description of the relevant step.|
```

File: src/guides/v2.3/migration/migration-tool-internal-spec.md [hunk 4]
```diff
@@ -165,6 +165,42 @@ Change prefix option in case Magento was installed with prefix in database table
...
+|integrity|It specifies the PHP class responsible for the integrity check. It compares the table field names, types, and other info to verify compatibility between Magento 1 and 2 data structures.|
```
Comment (BarnyShergold) [inline src/guides/v2.3/migration/migration-tool-internal-spec.md:173]:
```suggestion
|integrity|Specifies the PHP class responsible for the integrity check. Compares the table field names, types, and other info to verify compatibility between Magento 1 and 2 data structures.|
```

File: src/guides/v2.3/migration/migration-tool-internal-spec.md [hunk 5]
```diff
@@ -165,6 +165,42 @@ Change prefix option in case Magento was installed with prefix in database table
...
+|data|It specifies the PHP class responsible for the data check. It transfers the data, table by table from Magento 1 and 2.|
```
Comment (BarnyShergold) [inline src/guides/v2.3/migration/migration-tool-internal-spec.md:174]:
```suggestion
|data|Specifies the PHP class responsible for the data check. Transfers the data, table by table from Magento 1 to Magento 2.|
```
I believe it should say "from M1 to M2"

File: src/guides/v2.3/migration/migration-tool-internal-spec.md [hunk 6]
```diff
@@ -165,6 +165,42 @@ Change prefix option in case Magento was installed with prefix in database table
...
+|volume|It specifies the PHP class responsible for the volume check. It compares the number of records between tables to verify that the transfer was successful.|
```
Comment (BarnyShergold) [inline src/guides/v2.3/migration/migration-tool-internal-spec.md:175]:
```suggestion
|volume|Specifies the PHP class responsible for the volume check. Compares the number of records between tables to verify that the transfer was successful.|
```

File: src/guides/v2.3/migration/migration-tool-internal-spec.md [hunk 7]
```diff
@@ -165,6 +165,42 @@ Change prefix option in case Magento was installed with prefix in database table
...
+|delta|It specifies the PHP class responsible for the delta check. It transfers the delta from Magento 1 to Magento 2 after the full data migration.|
```
Comment (BarnyShergold) [inline src/guides/v2.3/migration/migration-tool-internal-spec.md:176]:
```suggestion
|delta|Specifies the PHP class responsible for the delta check. Transfers the delta from Magento 1 to Magento 2 after the full data migration.|
```

File: src/guides/v2.3/migration/migration-tool-internal-spec.md [hunk 8]
```diff
@@ -165,6 +165,42 @@ Change prefix option in case Magento was installed with prefix in database table
...
+
+### source available operations {#source}
```
Comment (BarnyShergold) [inline src/guides/v2.3/migration/migration-tool-internal-spec.md:178]:
```suggestion
### Source available operations {#source}
```

File: src/guides/v2.3/migration/migration-tool-internal-spec.md [hunk 9]
```diff
@@ -165,6 +165,42 @@ Change prefix option in case Magento was installed with prefix in database table
...
+
+|Document|Field|Required?|
+|--- |--- |--- |
```
Comment (BarnyShergold) [inline src/guides/v2.3/migration/migration-tool-internal-spec.md:181]:
```suggestion
|---|---|---|
```

File: src/guides/v2.3/migration/migration-tool-internal-spec.md [hunk 10]
```diff
@@ -165,6 +165,42 @@ Change prefix option in case Magento was installed with prefix in database table
...
+|name|Database name of the Magento 1 server.|yes|
+|host|Host IP address of the Magento 1 server|yes|
```
Comment (BarnyShergold) [inline src/guides/v2.3/migration/migration-tool-internal-spec.md:183]:
```suggestion
|host|Host IP address of the Magento 1 server.|yes|
```

File: src/guides/v2.3/migration/migration-tool-internal-spec.md [hunk 11]
```diff
@@ -165,6 +165,42 @@ Change prefix option in case Magento was installed with prefix in database table
...
+|port|Port number of the Magento 1 server.|no|
+|user|Username of the Magento 1 database server|yes|
```
Comment (BarnyShergold) [inline src/guides/v2.3/migration/migration-tool-internal-spec.md:185]:
```suggestion
|user|Username of the Magento 1 database server.|yes|
```

File: src/guides/v2.3/migration/migration-tool-internal-spec.md [hunk 12]
```diff
@@ -165,6 +165,42 @@ Change prefix option in case Magento was installed with prefix in database table
...
+|password|Password of the Magento 1 database server|yes|
```
Comment (BarnyShergold) [inline src/guides/v2.3/migration/migration-tool-internal-spec.md:186]:
```suggestion
|password|Password of the Magento 1 database server.|yes|
```

File: src/guides/v2.3/migration/migration-tool-internal-spec.md [hunk 13]
```diff
@@ -165,6 +165,42 @@ Change prefix option in case Magento was installed with prefix in database table
...
+|ssl_ca|Path to SSL Certificate Authority file|no|
```
Comment (BarnyShergold) [inline src/guides/v2.3/migration/migration-tool-internal-spec.md:187]:
```suggestion
|ssl_ca|Path to SSL Certificate Authority file.|no|
```

File: src/guides/v2.3/migration/migration-tool-internal-spec.md [hunk 14]
```diff
@@ -165,6 +165,42 @@ Change prefix option in case Magento was installed with prefix in database table
...
+|ssl_cert|Path to SSL Certificate file|no|
```
Comment (BarnyShergold) [inline src/guides/v2.3/migration/migration-tool-internal-spec.md:188]:
```suggestion
|ssl_cert|Path to SSL Certificate file.|no|
```

File: src/guides/v2.3/migration/migration-tool-internal-spec.md [hunk 15]
```diff
@@ -165,6 +165,42 @@ Change prefix option in case Magento was installed with prefix in database table
...
+|ssl_key|Path to SSL Key file|no|
```
Comment (BarnyShergold) [inline src/guides/v2.3/migration/migration-tool-internal-spec.md:189]:
```suggestion
|ssl_key|Path to SSL Key file.|no|
```

File: src/guides/v2.3/migration/migration-tool-internal-spec.md [hunk 16]
```diff
@@ -165,6 +165,42 @@ Change prefix option in case Magento was installed with prefix in database table
...
+
+### destination available operations {#destination}
```
Comment (BarnyShergold) [inline src/guides/v2.3/migration/migration-tool-internal-spec.md:191]:
```suggestion
### Destination available operations {#destination}
```

File: src/guides/v2.3/migration/migration-tool-internal-spec.md [hunk 17]
```diff
@@ -165,6 +165,42 @@ Change prefix option in case Magento was installed with prefix in database table
...
+
+|Document|Field|Required?|
+|--- |--- |--- |
```
Comment (BarnyShergold) [inline src/guides/v2.3/migration/migration-tool-internal-spec.md:194]:
```suggestion
|---|---|---|
```

File: src/guides/v2.3/migration/migration-tool-internal-spec.md [hunk 18]
```diff
@@ -165,6 +165,42 @@ Change prefix option in case Magento was installed with prefix in database table
...
+|name|Database name of the Magento 2 server.|yes|
```
Comment (BarnyShergold) [inline src/guides/v2.3/migration/migration-tool-internal-spec.md:195]: Put fullstop at the end of every field description statement

File: src/guides/v2.4/migration/migration-tool-internal-spec.md [hunk 19]
```diff
@@ -165,6 +165,42 @@ Change prefix option in case Magento was installed with prefix in database table
 
 Configuration data is accessible via \Migration\Config class.
```
Comment (BarnyShergold) [inline src/guides/v2.4/migration/migration-tool-internal-spec.md:167]: Repeat all corrections to the 2.3 version on the 2.4 version. Will double check 2.4

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Nice addition

**Review by BarnyShergold — APPROVED** [review]
Great addition!

File: src/guides/v2.3/migration/migration-tool-internal-spec.md [hunk 20]
```diff
@@ -165,6 +165,42 @@ Change prefix option in case Magento was installed with prefix in database table
 
 Configuration data is accessible via \Migration\Config class.
 
+### Steps available operations {#steps}
+
+|Document|Field|
+|---|---|
+|step|Second level node inside the Steps node. Description of the relevant step.|
```
Comment (dobooth) [inline src/guides/v2.3/migration/migration-tool-internal-spec.md:172]: We should make it clear that it is the title attribute that takes the description.

File: src/guides/v2.3/migration/migration-tool-internal-spec.md [hunk 21]
```diff
@@ -165,6 +165,42 @@ Change prefix option in case Magento was installed with prefix in database table
...
+|integrity|Specifies the PHP class responsible for the integrity check. Compares the table field names, types, and other info to verify compatibility between Magento 1 and 2 data structures.|
+|data|Specifies the PHP class responsible for the data check. Transfers the data, table by table from Magento 1 to Magento 2.|
+|volume|Specifies the PHP class responsible for the volume check. Compares the number of records between tables to verify that the transfer was successful.|
+|delta|Specifies the PHP class responsible for the delta check. Transfers the delta from Magento 1 to Magento 2 after the full data migration.|
+
+### Source available operations {#source}
+
+|Document|Field|Required?|
+|---|---|---|
+|name|Database name of the Magento 1 server.|yes|
```
Comment (dobooth) [inline src/guides/v2.3/migration/migration-tool-internal-spec.md:182]: These all go in the <database> node within the source and destination nodes. And these are not 'available operations' but really database information attributes.

**Review by dobooth — CHANGES_REQUESTED** [review]

**Comment by dobooth** [comment]
@ajithkumar-maragathavel ?

File: src/guides/v2.3/migration/migration-tool-internal-spec.md (hunk 20 above)
Comment (ajithkumar-maragathavel) [reply src/guides/v2.3/migration/migration-tool-internal-spec.md]: @dobooth  I have updated the description for this. Please review it and let me know If any changes requires

File: src/guides/v2.3/migration/migration-tool-internal-spec.md (hunk 21 above)
Comment (ajithkumar-maragathavel) [reply src/guides/v2.3/migration/migration-tool-internal-spec.md:182]: @dobooth  Thanks for the info. I have updated the title for this source and destination section. Please review it and let me know if any changes require.

**Review by dobooth — APPROVED** [review]

**Comment by dobooth** [comment]
@magento import code to magento-devdocs/devdocs

**Comment by magento-engcom-team** [comment]
@dobooth the branch with code successfully imported into`magento-devdocs/devdocs` repository. Branch name: `imported-magento-devdocs-8759`.

File: src/guides/v2.3/migration/migration-tool-internal-spec.md (hunk 6 above)
Comment (shap3e) [reply src/guides/v2.3/migration/migration-tool-internal-spec.md]: hi

**Pull request merged**